lib/metrics.coffee

```diff
@@ -48,6 +49,15 @@ module.exports =
       return if eventName of IgnoredCommands
       Reporter.sendCommand(eventName)
 
+    grim.on 'updated', (deprecation) ->
+      setImmediate ->
+        for path, stack of deprecation.stacks
+          packageName = stack.metadata?.packageName
+          pack = atom.packages.getLoadedPackage(packageName)
+          version = pack?.metadata?.version
+          Reporter.sendEvent('deprecation', packageName, version) if packageName?
```

Comment:
> Seems like core deps pass in no metadata, correct?

I was looking at this over the last few weeks and -- most deprecations don't have `metadata.packageName` defined, but some do because otherwise it's [hard to figure out which package caused the deprecation by looking at the call stack](https://github.com/atom/atom/pull/6510). Example of [one which does have `metadata.packageName`](https://github.com/atom/atom/blob/6f5e7cdd5c6dfd10e5d8ce740e9587f90c33fd2f/src/package.coffee#L286) and [one which doesn't](https://github.com/atom/atom/blob/4160a8c239f8a7ed54d2b6395c81a225748a8e98/src/project.coffee#L445).

Deprecation cop uses `metadata.packageName` if it exists, and if it doesn't -- it tries to figure out the package name by [digging through the stack](https://github.com/atom/deprecation-cop/blob/60c16962c27b076ec248626e6201322ffd8614d0/lib/deprecation-cop-view.coffee#L113-L133).